web3swift/src/Client/Models/EthereumAddress.swift

```diff
@@ -4,37 +4,61 @@
 //
 
 import Foundation
+import BigInt
 
 public struct EthereumAddress: Codable, Hashable {
-    public let value: String
+    @available(*, deprecated, message: "Shouldn't rely on the actual String representation. Use asString() instead to get an unformatted representation")
+    public var value: String {
+        raw
+    }
+    private let raw: String
```

Comment:
Did you consider making `EthereumAddress` conform to `RawRepresentable<String>`? That would give you a lot of the implementation for free.

You would only have to override the implementation for `Hashable` and `Equatable`.

Reply:
Just checked and it will require to expose `rawValue` as public. We really don't want to do that, and only expose formatting methods to force the caller to decide to which format to convert when getting the address